### ISSUE TITLE
Crash on Wayland when output refresh rate is zero

### ISSUE THREAD

**Issue by mstoeckl** [libsdl-org/SDL#4785]
SDL version: git master, b592e78f93b1e698b981ecf5af90e48e52e6387e; also with 2.0.16

When running an SDL program that uses OpenGL with the Wayland backend under a nested instance of Sway, the program crashes. (For example, from an X11 session , run `sway`; then inside Sway, run `SDL_VIDEODRIVER=wayland supertux2`.)

The stack trace implies the problem is that nested Sway sends a zero refresh rate, and then a division is performed using it. This is permitted by the [protocol specification](https://wayland.app/protocols/wayland#wl_output:event:mode). 
```
Program terminated with signal SIGFPE, Arithmetic exception.
#0  0x00007f06c7523b15 in Wayland_GLES_SwapWindow (_this=0x55936770f390, window=<optimized out>) at /usr/src/debug/SDL/src/video/wayland/SDL_waylandopengles.c:132
132	       const Uint32 max_wait = SDL_GetTicks() + (10000 / sdldisplay->current_mode.refresh_rate);  /* ~10 frames, so we'll progress even if throttled to zero. */
[Current thread is 1 (Thread 0x7f06c620d440 (LWP 21589))]
(gdb) bt
#0  0x00007f06c7523b15 in Wayland_GLES_SwapWindow (_this=0x55936770f390, window=<optimized out>) at /usr/src/debug/SDL/src/video/wayland/SDL_waylandopengles.c:132

(gdb) p *sdldisplay
$2 = {name = 0x5593677122f0 "wayland", max_display_modes = 32, num_display_modes = 1, display_modes = 0x559367711360, 
  desktop_mode = {format = 370546692, w = 851, h = 1383, refresh_rate = 0, driverdata = 0x5593677128c0}, 
  current_mode = {format = 370546692, w = 851, h = 1383, refresh_rate = 0, driverdata = 0x5593677128c0},
  orientation = SDL_ORIENTATION_UNKNOWN, fullscreen_window = 0x0, device = 0x55936770f390, driverdata = 0x559367712950}
```

Relevant part of Wayland protocol debug log:
```
[1965191.574] wl_output@18.geometry(0, 0, 0, 0, 0, "wayland", "wayland", 0)
[1965191.609] wl_output@18.mode(1, 1278, 1383, 0)
[1965191.628] wl_output@18.scale(1)
[1965191.635] wl_output@18.done()
```

**Comment by sulix** [comment]
I was able to reproduce this, and pull request #4788 fixes it (at least with sway nested under KWin wayland and supertux2) on my machine.

**Comment by flibitijibibo** [comment]
Fixed by #4788
